watcher.go

```diff
@@ -55,7 +60,10 @@ func WatchForUpdates(filename string, done <-chan bool, action func()) {
 				// can't be opened.
 				if event.Op&(fsnotify.Remove|fsnotify.Rename|fsnotify.Chmod) != 0 {
 					logger.Printf("watching interrupted on event: %s", event)
-					watcher.Remove(filename)
+					err = watcher.Remove(filename)
+					if err != nil {
+						logger.Printf("error removing watcher on %s: %s", filename, err)
```

Comment:
This was the last one I was torn on: do we crash or log when this operation fails?

Reply:
I think this one is a log because we are removing the internal watch process rather than doing anything else. I think we basically leak a goroutine here, not sure really what the effect would be though. Might be good to check the kinds of errors we can get out of this one